### DISCUSSION TITLE
Secondary Widgets limit

### DISCUSSION THREAD

**Discussion by zdunex25** [zepp-health/zeppos-forum#255] [Developer Feedback]
I am working on an app and I prepared 8 widgets, but I can only see 5 at the same time, to see different one I need to remove the other first, it does not affect built-in widgets I can add as many built-in widgets as I want.
Is there a limit for custom secondary widgets?
@MyoungXUE @silver-zepp

**Comment by MyoungXUE** [comment]
@zdunex25 

Hi, due to the memory consumption of the SecondaryWidget and the limited resources of the watch device, we have limited the maximum number to 5, please see our documentation for details.

<img width="732" alt="image" src="https://github.com/zepp-health/zeppos-forum/assets/107526244/76706866-f9ca-4f6c-bf14-1ae791333ed1">

https://docs.zepp.com/docs/guides/framework/device/secondary-widget/#zepp-os-20-open-related-js-api
